packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -191,7 +191,10 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
 
       if (this._connected) {
         // if the selected wallet is already connected, we don't need to connect again
-        if (this.wallet?.name === walletName) return;
+        if (this.wallet?.name === walletName)
+          throw new WalletConnectionError(
+            `${walletName} wallet is already connected`
+          ).message;
```

Comment:
What's up with the `.message` part of this?

Reply:
yea... this is to only print the `message` ( i.e `Petra wallet is already connected` ) instead of the whole object message which is `name`+`message` (i.e `WalletConnectionError: Petra wallet is already connected`). It relies on the built-in `Error` class.